### ISSUE TITLE
Nirspec IFU: Slit to MSA transform issue

### ISSUE THREAD

**Issue by nden** [STScI-JWST/jwst#1898]
The transform from slit to MSA for the IFU includes transforming slit coordinates
through the Slicer transforms, followed by the IFU-POST transforms.
Using Slity, lam supplied by ESA as inputs, the pipeline does not reproduce the MSA
positions outputs.
This issue is awaiting clarification by the IDT team.

**Comment by nden** [comment]
Update from the IDT team:

```
I had a look at our code to double check what we are writing out on the
extensions of the IFU traces and it is the "MSA - IFU entrance"
coordinates. I apologize if that is not clear from the extension
headers. To recap:

- in the calculation to find the trace and derive the wavelength, the
IFU calculations are done from the slicer to the direction of the FPA.
That MSA plane is what we call the *MSAout*, and the transforms to use
to go from slicer plane to MSAout are the IFU-POST[slice] (forward), as
described in my IFU note Sec. 3.1.2.

- in order to go to sky, the IFU calculations are done from the slicer
in the direction of V2V3. That MSA plane is what we call *MSAin* (where
light enters the IFU), and the transforms to use to go from slicer plane
to MSAin are the IFU-FORE^-1 (backward!), as described in my IFU note
Sec. 4.3 .

The values that we output in the Trace in the extensions called MSAx and
MSAy are the ones on the *MSAin* plane (the second example I gave
above). In your script you are calculating the MSAout. Once you have
MSAin, you are ready to go to OTEIP and then to V2V3.

On the bright side, I checked your example and if I calculate the MSAout
coordinates I get the same results as your script (machine precision
agreement!), great.
```

This means the code needs to be changed to remove the IFUPOST transform when going to sky.
There should be some indication that the `msa_frame` frame in the Nirspec IFU WCS pipeline refers to the MSA entrance (direction is from sky to detector). A description attribute can be used spacetelescope/gwcs#151

**Comment by nden** [comment]
@stscieisenhamer @jdavies-st The comments from the team clarify this issue.

**Comment by nden** [comment]
Fixed in commit [d0686a86c1](https://github.com/STScI-JWST/jwst/commit/d0686a86c11543fdaa17834b8427090ec9435cc4).
